libs/content/workspace/src/space_inspector/show.rs

```diff
@@ -0,0 +1,494 @@
+        }
+    }
+
+    pub fn change_root(&mut self, new_root: Uuid) {
+        self.data.focused_folder = new_root;
+        self.paint_order = vec![];
+    }
+
+    pub fn reset_root(&mut self) {
+        self.data.focused_folder = self.data.root;
+        self.paint_order = vec![];
+    }
+
+    pub fn get_color(&self, curr_id: Uuid, mut layer: usize, mut child_number: usize) -> Color32 {
+        let big_table = [
+            // red
+            [
+                Color32::from_rgb(128, 15, 47),
+                Color32::from_rgb(164, 19, 60),
+                Color32::from_rgb(201, 24, 74),
+                Color32::from_rgb(255, 77, 109),
+                Color32::from_rgb(255, 117, 143),
+                Color32::from_rgb(255, 143, 163),
+            ],
+            // green
+            [
+                Color32::from_rgb(27, 67, 50),
+                Color32::from_rgb(45, 106, 79),
+                Color32::from_rgb(64, 145, 108),
+                Color32::from_rgb(82, 183, 136),
+                Color32::from_rgb(116, 198, 157),
+                Color32::from_rgb(116, 198, 157),
+            ],
+            // blue
+            [
+                Color32::from_rgb(2, 62, 138),
+                Color32::from_rgb(0, 119, 182),
+                Color32::from_rgb(0, 150, 199),
+                Color32::from_rgb(0, 180, 216),
+                Color32::from_rgb(72, 202, 228),
+                Color32::from_rgb(144, 224, 239),
+            ],
+        ];
+        if layer == 1 {
+            if child_number > 2 {
+                child_number %= 3;
+            }
+            return big_table[child_number][0];
+        }
+
+        let parent_color = self
+            .colors
+            .iter()
+            .find(|item| item.id == self.data.all_files.get(&curr_id).unwrap().file.parent)
+            .unwrap()
+            .color;
+
+        let parent_type = big_table
+            .iter()
+            .enumerate()
+            .find_map(|(row_index, row)| {
+                row.iter()
+                    .position(|&x| x == parent_color)
+                    .map(|col_index| (row_index, col_index))
+            })
+            .unwrap()
+            .0;
+
+        layer -= 1;
+
+        if layer > 5 {
+            layer %= 6;
+        }
+
+        big_table[parent_type][layer]
+    }
+
+    pub fn follow_paint_order(
+        &mut self, ui: &mut Ui, root_anchor: Rect, window: Rect,
+    ) -> Option<Uuid> {
+        let mut changed_focused_folder: Option<Uuid> = None;
+        let mut current_layer = 0;
+        let mut current_position = root_anchor.min.x;
+        let mut child_number = 1;
+        let mut visited_folders: Vec<DrawHelper> = vec![];
+        let mut current_parent =
+            DrawHelper { id: self.data.focused_folder, starting_position: 0.0 };
+
+        for (i, item) in self.paint_order.iter().enumerate() {
+            let item_filerow = self.data.all_files.get(&item.id).unwrap();
```

Comment:
If you're down to reference `self.data.all_files` using `item.id` then why copy the file name into the cell? Just get the file name this way too